### PR TITLE
Rewrite the Finland (matka.fi) backend

### LINKED ISSUE #252: Add support for new matka.fi API
The Finnish travel planner matka.fi are replacing their current API, see http://developer.matka.fi/pages/en/home.php. The new API is documented at https://www.digitransit.fi/en/developers/. They write that the old API should be closed down at the end of 2016, but it still seems to work. Since it can stop working anytime now, the Fahrplan backend should be updated as soon as possible.

### PR DESCRIPTION
**Pull request by f03el**
A rewrite of the matka.fi backend for Finland, since they are closing down the old API.

Fixes #252.

### CODE REVIEW

**Comment by f03el** [comment]
@leppa @smurfy Any comments?

**Comment by smurfy** [comment]
Sorry, will have a look this weekend, quite busy at the moment

File: src/parser/parser_abstract.cpp [hunk 1]
```diff
@@ -145,6 +145,57 @@ QVariantMap ParserAbstract::parseJson(const QByteArray &json) const
...
+    return QJsonDocument(QJsonObject::fromVariantMap(doc)).toJson(QJsonDocument::Indented);
+}
+#else
+QByteArray toJson(const QVariant& value)
+{
+    if (value.isNull())
+        return "null";
+    switch (value.type()) {
+    case QVariant::String:
+        return "\"" + value.toString().replace("\"", "\\\"").toUtf8() + "\"";
```
Comment (leppa) [inline src/parser/parser_abstract.cpp]: There's, actually, much more that needs to be escaped, according to RFC. Unless you're sure about incoming values.

I wrote a simple `QVariant` to `JSON` converter in https://github.com/leppa/qtinappanalytics/blob/master/src/amplitudeanalytics/jsonfunctions_p.h. Feel free to use it, it's licensed under the BSD License.

Comment (f03el) [inline src/parser/parser_abstract.cpp]: I'm fine with using that instead. How do you suggest it to be included? Put the file in in the 3rdparty folder? It should only be included for Qt4 builds.

**Comment by f03el** [comment]
I got the email below today. The old API is now finally closed, so it would be good to have support for the new API added as soon as possible.

> Hello
> 
> As announced earlier (in October 2016)  the old Matka.fi API´s were officially closed down by end of 2016. The users were encouraged to shift to https://digitransit.fi/en/developers/. The old API’s are finally closed today 7.3.2017.
> 
> Best regards,
> 
> Matka.fi support

**Pull request merged**